File: apollo-compiler/src/test/graphql/com/example/mutation_create_review_semantic_naming/type/CustomType.java
package com.example.mutation_create_review_semantic_naming.type;

import com.apollographql.apollo.api.ScalarType;
import java.lang.Class;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import javax.annotation.Generated;

@Generated("Apollo GraphQL")
public enum CustomType implements ScalarType {
  DATE {
    @Override
    public String typeName() {
      return "Date";
    }

    @Override
    public Class javaType() {
      return Object.class;
    }
  },

  ID {
    @Override
    public String typeName() {
      return "ID";
    }

    @Override
    public Class javaType() {
      return String.class;
    }
  }
}

File: apollo-compiler/src/test/graphql/com/example/mutation_create_review_semantic_naming/type/CustomType.kt
package com.example.mutation_create_review_semantic_naming.type

import com.apollographql.apollo.api.ScalarType
import java.lang.Class
import javax.annotation.Generated
import kotlin.String

@Generated("Apollo GraphQL")
enum class CustomType : ScalarType {
    DATE {
        override fun typeName(): String = "Date"

        override fun javaType(): Class<*> = java.lang.Object::class.java
    },

    ID {
        override fun typeName(): String = "ID"

        override fun javaType(): Class<*> = String::class.java
    }
}
